src/controllers/expenses.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const expencesServices = require('../services/expenses');
+const userServices = require('../services/users');
+
+const getAll = (req, res) => {
+  const searchParams = req.query;
+
+  if (!searchParams) {
+    res.send(expencesServices.getAll());
+
+    return;
+  }
+
+  res.send(expencesServices.getFiltered(searchParams));
+};
+
+const getOne = (req, res) => {
+  const { expenseId } = req.params;
+  const foundExpense = expencesServices.getById(expenseId);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(foundExpense);
+};
+
+const add = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  const user = userServices.getById(userId);
+
+  const isCorrect = user && spentAt && title && amount && category;
```

Comment:
Maybe would be better to process a case if some fields won't be in the target type or a user with this userId doesn't exist